students/terrance_jones/lesson4/mailroom_part2.py

```diff
@@ -107,16 +98,17 @@ def create_report():
     heading = "{:<25s} | {:^20s} | {:^10s} | {:^20s} ".format("Donor Name", "Total Given", "Num Gifts", "Average Gift")
     print(heading)
     print("-" * len(heading))
-    print_row()
+    print_rows()
     menu_selection(sub_prompt, sub_dispatch)
 
 def letters_everyone():
-    for item in donor_list:
-        filename = "{}.txt".format(item['full_name'])
+    for key in donor_list.keys():
+        filename = "{}.txt".format(key)
         with open(filename, 'w') as f:
-            donor_letter = letter(item)
+            donor_letter = letter(key)
             f.write(donor_letter)
- 
+    print("Letters have been created for each donor")
+    menu_selection(sub_prompt, sub_dispatch)
```

Comment:
you shouldn't be calling these manually, you should have a `while True` loop at the main entry point that controls the main program flow.